OOP/LibraryManagement/LibraryApp/Services/LibraryService .cs

```diff
@@ -0,0 +1,36 @@
+﻿using LibraryApp.Documents;
+using LibraryApp.Repositories;
+
+namespace LibraryApp.Services;
+
+public class LibraryService<T>(IRepository<T> repository) : IService<T> where T : Document
```

Comment:
**Note**: In this scenario let's leave this implementation as it is.
However, in real commerce world app making of some generic services has more disadvantages than advantages.

Why? The reason is domain functionality requirements: your entities might be similar to each other, however, for some business goals it might have different prerequisites and if you are using just generic service, it will require constant updates that, probably, will lead to violation of such principles as single responsibility, open/close, etc.

So, my advise: for each entity (magazine, book, patent, etc.) there should be each unique service (like `MagazineService`, `BookService`, etc.) that contains its entity unique business logic features. Code duplication in this scenario is absolutely fine. If you don't want to see it, then you can create some abstract class like `BaseDocumentService` and place all reusable logic right there.